napari/layers/utils/layer_utils.py

```diff
@@ -204,81 +216,485 @@ def _nanmax(array):
     return max_value
 
 
-def calc_data_range(data, rgb=False) -> Tuple[float, float]:
+def calc_data_range(data, rgb: bool = False) -> None | Tuple[float, float]:
```

Comment:
```suggestion
def calc_data_range(data, rgb: bool = False) -> None | tuple[float, float]:
```

just simplify as we `from __future__ import annotations`